datafusion/sqllogictest/test_files/datetime/arith_time_interval.slt

```diff
@@ -0,0 +1,27 @@
+# time + interval → time
+# Add an interval to a time
+# time '01:00' + interval '3 hours' → 04:00:00
+
+query ?
+SELECT '01:00'::time + interval '3 hours'
+----
+4 hours
+
+query T
+SELECT arrow_typeof('01:00'::time + interval '3 hours')
+----
+Interval(MonthDayNano)
+
+# time - interval → time
+# Subtract an interval from a time
+# time '05:00' - interval '2 hours' → 03:00:00
+
+query ?
+SELECT '05:00'::time - interval '2 hours'
```

Comment:
Please add cases for time-interval < 0 hours, e.g. SELECT '02:00'::time - interval '3 hours'